dra-evolution/pkg/api/capacity_types.go

```diff
@@ -1,170 +1,162 @@
-	// resources.
-	//
-	// +optional
-	ClaimResourcesProvided []ResourceCapacity `json:"claimResourcesProvided,omitempty"`
+	// TODO for 1.31: define how to support partitionable devices
 }
 
-type ResourceCapacity struct {
-	// Name is the resource name/type.
-	// +required
-	Name string `json:"name"`
-
-	// Capacity is the total capacity of the named resource.
-	// +required
-	Capacity resource.Quantity `json:"capacity"`
-
-	// BlockSize is the increments in which capacity is consumed. For
-	// example, if you can only allocate memory in 4k pages, then the
-	// block size should be "4Ki". Default is 1.
+// DeviceAttribute is a combination of an attribute name and its value.
+type DeviceAttribute struct {
+	// Name is a unique identifier across all possible attributes of devices.
 	//
-	// If the resource is consumable in a fractional way, then the
-	// default of 1 should not be used; instead this should be a fractional
-	// amount corresponding the increment size. We may also need a minimum
-	// value, if the minimum is larger than the block size (as is the case
-	// for standard Kubernetes CPU resources).
+	// If this is a DNS subdomain (no dot), then the driver name gets added
+	// when looking up attributes. This avoids name collisions with attributes
+	// used by other drivers.
 	//
-	// +optional
-	BlockSize *resource.Quantity `json:"blockSize,omitempty"`
-}
-
-// Attribute capture the name, value, and type of an device attribute.
-type Attribute struct {
-	Name string `json:"name"`
+	// If this is a full DNS domain, then the meaning of the attribute driver-independent.
+	// For example, Kubernetes will use `*.k8s.io` names when defining attributes that
+	// drivers from different vendors are supposed to use.
+	Name string `json:"name" protobuf:"bytes,1,name=name"`
 
-	// One of the following:
-	StringValue   *string            `json:"stringValue,omitempty"`
-	IntValue      *int               `json:"intValue,omitempty"`
-	QuantityValue *resource.Quantity `json:"quantityValue,omitempty"`
-	SemVerValue   *SemVer            `json:"semVerValue,omitempty"`
+	DeviceAttributeValue `json:",inline" protobuf:"bytes,2,opt,name=attributeValue"`
 }
 
-func (a Attribute) Equal(b Attribute) bool {
-	if a.Name != b.Name {
-		return false
-	}
-
-	return a.EqualValue(b)
+// The Go field names below have a Value suffix to avoid a conflict between the
+// field "String" and the corresponding method. That method is required.
+// The Kubernetes API is defined without that suffix to keep it more natural.
+
+// DeviceAttributeValue must have one and only one field set.
+type DeviceAttributeValue struct {
+	// QuantityValue is a quantity.
+	QuantityValue *resource.Quantity `json:"quantity,omitempty" protobuf:"bytes,6,opt,name=quantity"`
+	// BoolValue is a true/false value.
+	BoolValue *bool `json:"bool,omitempty" protobuf:"bytes,2,opt,name=bool"`
+	// IntValue is a 64-bit integer.
+	IntValue *int64 `json:"int,omitempty" protobuf:"varint,7,opt,name=int"`
```

Comment:
Why do we need Quantity and Int?

Reply:
I suppose we can drop Int in favor of Quantity.

Reply:
We can always add it back if we have a real reason